### ISSUE TITLE
[bug]: channel stuck in pending state

### ISSUE THREAD

**Issue by maxrantil** [lightningnetwork/lnd#7275]
### Background
I've been trying to open a channel through Thunberhub for six weeks, but the transaction never made it into the mempool and has been stuck in a pending state since then. I tried closing the channel yesterday using the 'lncli closechannel' command with the '--force' flag, but this didn't work either. The state changed from 'pending open' to 'pending close', but the transaction still didn't make it into the mempool. I've tried restarting lnd and the node, scanning the wallet, and looking through the logs, but none of these actions have fixed the problem. I've also sought help from knowledgeable people in Telegram groups, but they suggested I open an issue. There is a significant amount of sats in limbo and I would greatly appreciate some help resolving this issue.

### Your environment

* LND 0.15.5-beta
* Linux umbrel 5.10.103-v8+  ... aarch64 GNU/Linux
* Bitcoin Core 24.0.1

### Expected behaviour

I have successfully opened many channels before, so I am not sure how to reproduce this issue. I have always followed the same process when opening channels, so I am not sure what might be causing this problem.

### Actual behaviour

When I run the 'lncli pendingchannels' command, I am able to see information about the channel, but none of the transaction IDs displayed are in the mempool.

**Comment by yyforyongyu** [comment]
So the channel has been stuck for 6 weeks? What's the funding outpoint?

**Comment by maxrantil** [comment]
Yes exactly. The channel point is: 

"channel point": " 824214c2f39d85611fc83896bc739e3f76c0f5b5fe957eed8d5340281d6a60e:1".

**Comment by Roasbeef** [comment]
No, we want the funding outpoint, or the txid of the transaction that created (funded) the channel.

**Comment by PCAanalytics** [comment]
I have the same issue I believe. When it attempted to open this channel, the transaction was never recorded in the mempool. So its sort of in limbo on my node.

I entered command to force close it as it had been pending for weeks. Now RTL shows it is closing the channel. That process was initiated 2 days ago, but RTL still says the following:

Wait Closing Channel Information
Transaction ID
03e53180a88e8765a8d386c010018afa4544c5a5aee48baefbff1953429721af
Channel Point
53aef06e5df7cc9865774ac02d665143e85bd9c8ac5353c2eb80fe8ddb80ae09:1
Peer Alias
CentralLondonBlitz
Peer Node Pubkey
03e472ddc37b014ce7d5f942b00b28fb1db53d461c4a89173962abd4b331a85179
Capacity
1,000,000
Limbo Balance
996,616
Local Balance
996,616
Remote Balance
0

**Comment by maxrantil** [comment]
Thanks guys for showing interesed in helping out, please bare with me for now knowing exactly what you ask for. Is this what you are asking me for?

"channel_point": "f824214c2f39d85611fc83896bc739e3f76c0f5b5fe957eed8d5340281d6a60e:1",

This is not in the mempool as I said but maybe you can get something out for it anyway? If its not the thing you are asking for please tell me how to get the funding outpoint, or the txid.

**Comment by PCAanalytics** [comment]
I read somewhere that an pending / closing channel can be manually deleted from the local node database. If so, can someone at Lightning please let me know how to do it. This channel transaction appears to be isolated on my local node.

Surely there should be a way to close a channel that never reached the mempool? I started the close process last Tuesday, but its still 'Wait Closing Channel Information'.

If it is waiting to get this information from the remote node, in my case CentralLondonBlitz, surely it will be waiting forever?

It is currently interfering with the operation of my node.

**Comment by PCAanalytics** [comment]
I notice from this https://github.com/ElementsProject/lightning/issues/1522, that it takes 2 weeks to close. So I wait.

**Comment by guggero** [comment]
@maxrantil yes, that is the channel point we were asking for. And it does look like the transaction isn't in the mempool.
Can you check your list of transactions (`lncli listchaintxns`) if you see `f824214c2f39d85611fc83896bc739e3f76c0f5b5fe957eed8d5340281d6a60e` somewhere? And if yes, post it here?

**Comment by guggero** [comment]
@PCAanalytics if you opened the channel, it won't disappear after two weeks. Can you check if the funding transaction is still not in the mempool? And see what inputs it used to spend? There is the `lncli abandonchannel` command. But that is only safe to be used if you are certain the funding transaction can never be confirmed (e.g. by making sure all its inputs were spent otherwise).

**Comment by PCAanalytics** [comment]
@guggero The transaction for opening this channel cant be found in the mempool. It was in the pending state for weeks / months until I issued the force close command.

On RTL it says:

Wait Closing Channel Information
Transaction ID
03e53180a88e8765a8d386c010018afa4544c5a5aee48baefbff1953429721af
Channel Point
53aef06e5df7cc9865774ac02d665143e85bd9c8ac5353c2eb80fe8ddb80ae09:1
Peer Alias
CentralLondonBlitz
Peer Node Pubkey
03e472ddc37b014ce7d5f942b00b28fb1db53d461c4a89173962abd4b331a85179
Capacity
1,000,000
Limbo Balance
996,616
Local Balance
996,616
Remote Balance
0


I did a lncli listpeers and see the following info about this problematic channel:

"pub_key": "03e472ddc37b014ce7d5f942b00b28fb1db53d461c4a89173962abd4b331a85179",
            "address": "a5cxqhkakdytx46w2jmnjh6x4cf2ru42odoadgcyho3h6atfafq3izyd.onion:9735",
            "bytes_sent": "63440",
            "bytes_recv": "1160097",
            "sat_sent": "0",
            "sat_recv": "0",
            "inbound": false,
            "ping_time": "929953",
            "sync_type": "PASSIVE_SYNC",
            "features": {
                "0": {
                    "name": "data-loss-protect",
                    "is_required": true,
                    "is_known": true
                },
                "5": {
                    "name": "upfront-shutdown-script",
                    "is_required": false,
                    "is_known": true
                },
                "7": {
                    "name": "gossip-queries",
                    "is_required": false,
                    "is_known": true
                },
                "9": {
                    "name": "tlv-onion",
                    "is_required": false,
                    "is_known": true
                },
                "12": {
                    "name": "static-remote-key",
                    "is_required": true,
                    "is_known": true
                },
                "14": {
                    "name": "payment-addr",
                    "is_required": true,
                    "is_known": true
                },
                "17": {
                    "name": "multi-path-payments",
                    "is_required": false,
                    "is_known": true
                },
                "19": {
                    "name": "wumbo-channels",
                    "is_required": false,
                    "is_known": true
                },
                "23": {
                    "name": "anchors-zero-fee-htlc-tx",
                    "is_required": false,
                    "is_known": true
                },
                "27": {
                    "name": "shutdown-any-segwit",
                    "is_required": false,
                    "is_known": true
                },
                "31": {
                    "name": "amp",
                    "is_required": false,
                    "is_known": true
                },
                "45": {
                    "name": "explicit-commitment-type",
                    "is_required": false,
                    "is_known": true
                },
                "2023": {
                    "name": "script-enforced-lease",
                    "is_required": false,
                    "is_known": true
                }
            },
            "errors": [
            ],
            "flap_count": 46,
            "last_flap_ns": "1673050092114298137",
            "last_ping_payload": "0000002043e8f90c2353c88a29851eef9db49cc1c2a467af6ff7010000000000000000005f0128bf323a3994481d92ddc57fd4c85f7800c73b9994302d917c2e16c312d7aa3fb9637e4108178803f056"
        }

**Comment by PCAanalytics** [comment]
No can't find it in the list of transactions.

**Comment by guggero** [comment]
Hmm, then I guess there's no easy way to check if all the inputs were spent. So I'd get rid of the channel with `lncli abandonchannel 53aef06e5df7cc9865774ac02d665143e85bd9c8ac5353c2eb80fe8ddb80ae09 1 --i_know_what_i_am_doing` and then to be on the safe side send all your on-chain funds to a new address in your wallet (to make sure all inputs are spent and the funding TX can never confirm in any case).

**Comment by maxrantil** [comment]
@guggero , I couldnt find it in the ``listchaintxns``netiher. You reckon I do the same and ``lncli abandonchannel`` and send the funds so a new address?

**Comment by guggero** [comment]
Yes. I'd rather make sure that everything is safe by taking a look at the funding TX. But if it's not around anymore, sending all funds to a new address is the second best you can do. I assume you restarted lnd recently (which attempts to re-broadcast the funding tx)? Did you see "Removing invalid transaction" in the logs?

**Comment by maxrantil** [comment]
Ok thanks, I have restarted it and looked carefully in the logs, but havn't seen "Removing invalid transaction" there. 

just did the ``abandonchannel`` and it is now closed and not in ``lncli pendingchannels`` anymore, im not sure if the fund found its way back tho. Im starting to realize that im a bit out on too deep waters. I really appreciate your help! But im going to play with a little less sats from now on.
